src/util/sighash.rs

```diff
@@ -36,6 +36,14 @@ use crate::{Script, Transaction, TxOut};
 
 use super::taproot::LeafVersion;
 
+/// Used for signature hash for invalid use of SIGHASH_SINGLE.
+pub(crate) const UINT256_ONE: [u8; 32] = [
```

Comment:
Perhaps this should be defined in the `constants` module and made public as a courtesy because everyone that uses `encode_signing_data_to` is going to have to define this, right? When I defined this array I had to look up what was supposed to be returned, then work out what a "one array" looks like. So, at least for me, it took a little while to do. We could then point users towards it in the function comment.

Reply:
Should I rename it something more meaningful or is the current name OK?

Reply:
If you can come up with a better name by all means rename it, that was the best I could do at the time :)